MonoXAnalysis/scripts/copyTreesFromEOSToDcache.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+
+import os
+import CMGTools.Production.eostools as eostools
+from optparse import OptionParser
+
+if __name__ == "__main__":
+
+    usage="""%prog [options] remoteEOSdir remoteDCACHEdir
+
+# This script can be used to copy files from EOS to DCache in Rome T2.
+#
+# It needs a valid grid proxy
+
+"""
+
+    parser = OptionParser(usage=usage)
+    parser.add_option("-s", "--select", dest="select", type='string', default=None, help='Filter on files with this pattern')
+    parser.add_option("-r", "--remove", dest="remove", type='string', default=None, help='Do not copy files with this pattern')
+    parser.add_option("-t", "--destt2", dest="destT2", type='string', default="T2_IT_Rome", help='destination T2')
+    parser.add_option("--dryrun", dest="dryrun", action="store_true", default=False, help='only print the commands, do not run the copy')
+    (options, args) = parser.parse_args()
+    
+    if len(args)<2: raise RuntimeError, 'Expecting at least two arguments'
+    
+    eosdir = args[0]
+    remdir = args[1]
+    
+    if not eostools.isEOS(eosdir): raise RuntimeError, 'Remote directory should be on EOS.'
+
+    files = eostools.ls(eosdir)
+    if options.select: files = [f for f in files if options.select in f]
+    if options.remove: files = [f for f in files if options.remove not in f]
+
+    tocopy = []
+    for f in files:
+        basename = os.path.basename(f)
+        tocopy.append( basename )
+        print "Will copy file ",basename
+
+    print '\nCopy the above files from %s -> %s. Do you agree? [y/N]\n' % (eosdir,remdir)
+    if raw_input()!='y':
+        print 'Aborting'
+        exit()
+
+    command = 'lcg-cp -b --vo cms -D srmv2 -U srmv2 -v'
+    eosconn='srm://srm-eoscms.cern.ch:8443/srm/v2/server?SFN='
+    remconn=''
+    if options.destT2=="T2_IT_Rome": remconn='srm://cmsrm-se01.roma1.infn.it:8443/srm/managerv2?SFN='
+    else:
+        print 'Only T2_IT_Rome implemented. Exiting.'
```

Comment:
You can use phedex to figure out that magic string for any T2.
E.g. if you do a query for ```https://cmsweb.cern.ch/phedex/datasvc/json/prod/lfn2pfn?protocol=srm&node=T2_IT_Rome&lfn=/store/foo``` you get a json which contains ```"pfn":"srm://cmsrm-se01.roma1.infn.it:8443/srm/managerv2?SFN=/pnfs/roma1.infn.it/data/cms/store/foo"```.